accepted/future-releases/class-modifiers/feature-specification.md

```diff
@@ -530,62 +576,142 @@ mixinModifier     ::= 'sealed' | 'base' | 'interface' | 'final'
 
 ## Static semantics
 
+A pair of definitions:
+
+*   A *pre-feature library* is a library whose language version is lower than
+    the version this feature is released in.
+
+*   A *post-feature library* is a library whose language version is at or above
+    the version this feature is released in.
+
+### Basic restrictions
+
 It is a compile-time error to:
 
-*   Extend a class marked `interface`, `final` or `sealed` outside of the library
-    where it is declared.
-    
-*   Implement the interface of a class or mixin marked `base`, `final` or `sealed`
-    outside of the library where it is declared.
-    
-*   Mix in a mixin or mixin class marked `interface`, `final` or `sealed` outside 
-    of the library where it is declared.
+*   Extend a class marked `interface`, `final` or `sealed` outside of the
+    library where it is declared.
     
-*   Extend a class marked `base` outside of the library where it is declared
-    unless the extending class is marked `base` or `final`. *This ensures that a
-    subtype can't escape the `base` restriction of its supertype by offering its
-    _own_ interface that could then be implemented without inheriting the
-    concrete implementation from the supertype.* <!-- Needs to account for `sealed` -->
+*   Implement the interface of a class or mixin marked `base`, `final` or
+    `sealed` outside of the library where it is declared.
     
-*   Mix in a mixin or mixin class marked `base` outside of the library where it
-    is declared unless the class mixing it in is marked `base` or `final`. *As
-    with the previous rule, ensures you can't get a backdoor interface on a
-    mixin that doesn't want to expose one.*
-    
-*   Apply `mixin` to a class whose superclass is not `Object` or that declares a
-    _non-trivial generative constructor_.
-    *Such a class can have an `extends` clause of the form `extends Object`,
-    or no `extends` clause. It cannot have any `with` clause.*
+*   Mix in a mixin or mixin class marked `interface`, `final` or `sealed`
+    outside of the library where it is declared.
+
+There is no direct restriction on types allowed in `on` clauses of mixin
```

Comment:
Yes, `B` is an error because it's not a `base` declaration, and `C` is not an error.

This ensures that no subtype of `A` has an implementable interface.
Another rule ensures, effectively, that other libraries also cannot ignore that `base` within their own library, when it was forced by another library.

Examples is a good idea.

Reply:
Added examples. Just removed the sentence since I think it distracts instead of clarifying.

It's like telling someone there's no alligator in the pantry. They probably weren't even thinking about alligators and now they're wondering if they should be.